dgraph/cmd/zero/http.go

```diff
@@ -256,22 +256,29 @@ func (s *Server) zeroHealth(ctx context.Context) (*api.Response, error) {
 }
 
 func (st *state) pingResponse(w http.ResponseWriter, r *http.Request) {
-	var err error
 	x.AddCorsHeaders(w)
 
 	switch r.Header.Get("Accept") {
 		case "application/json":
-			var resp *api.Response
-			if resp, err = (st.zero).zeroHealth(r.Context()); err != nil {
+			resp, err := (st.zero).zeroHealth(r.Context())
+			if err != nil {
 				x.SetStatus(w, x.Error, err.Error())
 				return
 			}       
 			w.Header().Set("Content-Type", "application/json; charset=utf-8")
 			w.WriteHeader(http.StatusOK)
-			_, _ = w.Write(resp.Json)
+			_, err = w.Write(resp.Json)
```

Comment:
I think we should just log (warning) the error. Good chance, if write has failed, something is really wrong with the request and SetStatus may not work.